src/compiler/core.ts

```diff
@@ -2197,13 +2200,13 @@ namespace ts {
         return result;
     }
 
-    export function cartesianProduct<T>(arrays: readonly T[][]) {
+    export function cartesianProduct<T>(arrays: ReadonlyArray<T>[]) {

```

Comment:
Never mind, I see that the rule does handle `readonly`. We might want to use `array-simple`.

Reply:
@uniqueiniquity `array-type` option `{ default: "array", readonly: "array" }` throws error for `ReadonlyArray` and suggest to use `readonly` instead of `ReadonlyArray`. The problem that there are many places where `ReadonlyArray` is used, and if we enable this option - PR will contain a lot of changes. This is not a big problem, just need _confirmation_ to make that change.

Reply:
Chatted with @RyanCavanaugh - let's do `{ default: "array", readonly: "array" }`.
Worth mentioning that `readonly T[][]` means the opposite of `ReadOnlyArray<T>[]`.
In the former, the outer array is readonly, in the latter it's the inner.

Reply:
@uniqueiniquity changed